### PR TITLE
Search Conf Incorrect successful alert message

### LINKED ISSUE #21867: BO - Search - Incorrect successful alert message when update 'Search'
#### Describe the bug

In the BO > Search page, when we try to update an alias, the successful message is displayed.
'Creation successful' instead of 'Update successful'
image

#### Expected behavior

The successful message should be 'Update successfull'

#### Steps to Reproduce

1. Go to 'BO>Shop Parameters > Search'
2. Click on 'Edit' alias
3. Edit and save
4. See error 'Incorrect message'

**Screenshots**

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.8.0
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | n the BO > Search page, when we try to update an alias, the successful message is displayed. Creation successful instead of Update successful
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/21867
| How to test?      | Check the wording
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24352)

### CODE REVIEW

File: controllers/admin/AdminSearchConfController.php [hunk 1]
```diff
@@ -472,7 +472,11 @@ public function processSave()
         }
 
         if (empty($this->errors)) {
-            $this->confirmations[] = $this->trans('Creation successful', [], 'Admin.Shopparameters.Notification');
+            if (Tools::getValue('id_alias')) {
+                $this->confirmations[] = $this->trans('Successful update.', [], 'Admin.Notifications.Success');
+            } else {
+                $this->confirmations[] = $this->trans('Successful creation.', [], 'Admin.Notifications.Success');
+            }
```
Comment (PululuK) [inline controllers/admin/AdminSearchConfController.php]:
```suggestion
            if (Tools::getValue('id_alias')) {
                $this->confirmations[] = $this->trans('Update successfull', [], 'Admin.Notifications.Success');
            } else {
                $this->confirmations[] = $this->trans('Creation successful', [], 'Admin.Notifications.Success');
            }
```

**Review by PululuK — DISMISSED** [review]
@okom3pom LGTM ! 
Just improve wording.

File: controllers/admin/AdminSearchConfController.php (hunk 1 above)
Comment (ghost) [reply controllers/admin/AdminSearchConfController.php]: Hi @PululuK it's beacause this wording already exist at the same place ( Admin.Notifications.Success ) but i can use : 

Update successful in `Admin.Notifications.Success`

And 

Creation successful in `Admin.Shopparameters.Notification
`
The translations are not at the same place.

Comment (ghost) [reply controllers/admin/AdminSearchConfController.php]: Or i add a new wording ?

Comment (PululuK) [reply controllers/admin/AdminSearchConfController.php]: ping @Julievrz

File: controllers/admin/AdminSearchConfController.php [hunk 2]
```diff
@@ -472,7 +472,11 @@ public function processSave()
         }
 
         if (empty($this->errors)) {
-            $this->confirmations[] = $this->trans('Creation successful', [], 'Admin.Shopparameters.Notification');
+            if (Tools::getValue('id_alias')) {
+                $this->confirmations[] = $this->trans('Successful update.', [], 'Admin.Notifications.Success');
```
Comment (Julievrz) [inline controllers/admin/AdminSearchConfController.php]:
```suggestion
                $this->confirmations[] = $this->trans('Update successful', [], 'Admin.Notifications.Success');
```

File: controllers/admin/AdminSearchConfController.php [hunk 3]
```diff
@@ -472,7 +472,11 @@ public function processSave()
...
+            } else {
+                $this->confirmations[] = $this->trans('Successful creation.', [], 'Admin.Notifications.Success');
```
Comment (Julievrz) [inline controllers/admin/AdminSearchConfController.php]:
```suggestion
                $this->confirmations[] = $this->trans('Successful creation', [], 'Admin.Notifications.Success');
```
This new string (without the period and in Admin.Notifications.Success) was introduced in other PRs so let's stick with this one. :-)

**Comment by ghost** [comment]
Thank you @PululuK && @Julievrz

**Review by PululuK — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi there !

I think the problem is a bit more complex for this page, because the "Edit alias" can give 3 different results depending on what's modified in the form, and I'm not sure the correct message should always be "Update successful".

Ping @PrestaShop/product-team , need some help for this one !

- **Case 1 :** 

I edit only the "Result" field : In this case, it's indeed only an update of the alias, no new alias is created. For this case, the message is OK.
In the DB, the alias is updated too.

Edit page : 
pr 24352 result mod 1

Result : 
pr 24352 result mod 2

- **Case 2 :** 

I edit both "Alias" and "Result" field : In this case, a new alias is created in addition to the original one. IMHO, the message is not OK : It should be "Successful creation". Or the behavior should be modified and the original alias should be replaced by the new one.
In DB, a new alias is indeed created and the original one is not modified.

Edit page : 
pr 24352 alias result mod 1

Result : 
pr 24352 alias result mod 2


- **Case 3 :** 

I edit only the "Alias" field : In this case, a new alias is created in addition to the original one, but if I edit the same alias again, both aliases are displayed in the "Alias" field. In this case, I'm not sure what should be the message displayed...
In DB, a new alias is created and the original one is not modified.

Edit page : 
pr 24352 alias mod 1

Result : 
pr 24352 alias mod 2

Both aliases in one edit page :
pr 24352 alias mod 3

I made a screenrecord of these 3 cases : 

https://drive.google.com/file/d/1GYGQw6sCNiFH7ZThWw6xRklJxi7yc1Eu/view?usp=sharing

Thanks :)

**Comment by MatShir** [comment]
There is a problem in the behavior as @Robin-Fischer-PS pointed out. In any case, this is an update workflow, so the message is correct even if the actions lead to the creation of a new alias. The behavior is not new, I could reproduce the problem in 1.7.6. So we are not going to adapt the message to a faulty behavior, but we should correct the behavior to match the action and follow the message. 
The PR is good, but fixing the behavior should be the subject of another PR.

**Comment by Robin-Fischer-PS** [comment]
Thanks @MatShir !

Then this PR is OK : 
Display of the confirmation message on edit : :heavy_check_mark: 
Display of the confirmation message on creation : :heavy_check_mark: 

It's QA :heavy_check_mark: ! Thanks @okom3pom :)

**Comment by PululuK** [comment]
Thanks @okom3pom @Robin-Fischer-PS

**Pull request merged**